### ISSUE TITLE
Structural updates that make invalid PointInstancers valid do not refresh imaging properly

### ISSUE THREAD

**Issue by asluk** [PixarAnimationStudios/USD#1076]
If the stage contains an invalid PointInstancer, and we author an update to make the PointInstancer valid, imaging does not reflect the change.

The repro cases I will attach have an instancer with missing prototypes, and an instancer with misssing positions.

This reproduces both in `master` and `dev`, currently at `ad19ac016c7984273305252caeee0431c8548419`.

We have a fix.

**Comment by asluk** [comment]
# repro-prototypes.usda

```#usda 1.0
(

)

def PointInstancer "pointinstancer"
{
    int[] protoIndices = [0, 1]
    
    rel prototypes = [
        ]

    rel prototypes_backup = [
        </pointinstancer/mesh0>,
        </pointinstancer/mesh1>,
        ]
		
    def Mesh "mesh0"
    {
        uniform bool doubleSided = 1
        float3[] extent = [(-0.1, -0.09510565, -0.125), (0.1, 0.09510566, 0.125)]
        int[] faceVertexCounts = [3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3]
        int[] faceVertexIndices = [0, 2, 1, 0, 3, 2, 0, 4, 3, 0, 5, 4, 0, 6, 5, 0, 7, 6, 0, 8, 7, 0, 9, 8, 10, 12, 11, 10, 13, 12, 14, 16, 15, 14, 17, 16, 18, 20, 19, 18, 21, 20, 22, 24, 23, 22, 25, 24, 26, 28, 27, 26, 29, 28, 30, 32, 31, 30, 33, 32, 34, 36, 35, 34, 37, 36, 38, 40, 39, 38, 41, 40, 42, 44, 43, 42, 45, 44, 42, 46, 45, 42, 47, 46, 42, 48, 47, 42, 49, 48, 42, 50, 49, 42, 51, 50, 52, 54, 53, 52, 55, 54, 56, 58, 57, 56, 59, 58]
        point3f[] points = [(-0.0809017, -0.058778536, -0.125), (-0.1, -8.742278e-9, -0.125), (-0.080901705, 0.05877852, -0.125), (-0.030901704, 0.09510565, -0.125), (0.030901698, 0.09510566, -0.125), (0.080901705, 0.058778524, -0.125), (0.1, 0, -0.125), (0.080901735, -0.058778495, -0.125), (0.030901713, -0.09510565, -0.125), (-0.03090171, -0.09510565, -0.125), (-0.080901705, 0.05877852, -0.125), (-0.1, -8.742278e-9, -0.125), (-0.1, -8.742278e-9, 0.125), (-0.080901705, 0.05877852, 0.125), (-0.030901704, 0.09510565, 0.125), (0.030901698, 0.09510566, 0.125), (0.030901698, 0.09510566, -0.125), (-0.030901704, 0.09510565, -0.125), (-0.030901704, 0.09510565, 0.125), (-0.030901704, 0.09510565, -0.125), (-0.080901705, 0.05877852, -0.125), (-0.080901705, 0.05877852, 0.125), (0.080901705, 0.058778524, -0.125), (0.030901698, 0.09510566, -0.125), (0.030901698, 0.09510566, 0.125), (0.080901705, 0.058778524, 0.125), (0.080901705, 0.058778524, -0.125), (0.080901705, 0.058778524, 0.125), (0.1, 0, 0.125), (0.1, 0, -0.125), (0.080901735, -0.058778495, 0.125), (0.030901713, -0.09510565, 0.125), (0.030901713, -0.09510565, -0.125), (0.080901735, -0.058778495, -0.125), (0.080901735, -0.058778495, 0.125), (0.080901735, -0.058778495, -0.125), (0.1, 0, -0.125), (0.1, 0, 0.125), (-0.0809017, -0.058778536, -0.125), (-0.0809017, -0.058778536, 0.125), (-0.1, -8.742278e-9, 0.125), (-0.1, -8.742278e-9, -0.125), (-0.03090171, -0.09510565, 0.125), (0.030901713, -0.09510565, 0.125), (0.080901735, -0.058778495, 0.125), (0.1, 0, 0.125), (0.080901705, 0.058778524, 0.125), (0.030901698, 0.09510566, 0.125), (-0.030901704, 0.09510565, 0.125), (-0.080901705, 0.05877852, 0.125), (-0.1, -8.742278e-9, 0.125), (-0.0809017, -0.058778536, 0.125), (-0.03090171, -0.09510565, 0.125), (-0.0809017, -0.058778536, 0.125), (-0.0809017, -0.058778536, -0.125), (-0.03090171, -0.09510565, -0.125), (-0.03090171, -0.09510565, 0.125), (-0.03090171, -0.09510565, -0.125), (0.030901713, -0.09510565, -0.125), (0.030901713, -0.09510565, 0.125)]
        uniform token subdivisionScheme = "none"
    }

    def Mesh "mesh1"
    {
        uniform bool doubleSided = 1
        float3[] extent = [(-0.25, -0.23776412, -0.275), (0.25, 0.23776415, 0.275)]
        int[] faceVertexCounts = [3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3]
        int[] faceVertexIndices = [0, 2, 1, 0, 3, 2, 0, 4, 3, 0, 5, 4, 0, 6, 5, 0, 7, 6, 0, 8, 7, 0, 9, 8, 10, 12, 11, 10, 13, 12, 14, 16, 15, 14, 17, 16, 18, 20, 19, 18, 21, 20, 22, 24, 23, 22, 25, 24, 26, 28, 27, 26, 29, 28, 30, 32, 31, 30, 33, 32, 34, 36, 35, 34, 37, 36, 38, 40, 39, 38, 41, 40, 42, 44, 43, 42, 45, 44, 46, 48, 47, 46, 49, 48, 50, 52, 51, 50, 53, 52, 50, 54, 53, 50, 55, 54, 50, 56, 55, 50, 57, 56, 50, 58, 57, 50, 59, 58]
        point3f[] points = [(0.20225425, 0.14694631, -0.275), (0.25, 0, -0.275), (0.20225431, -0.14694624, -0.275), (0.07725428, -0.23776412, -0.275), (-0.07725427, -0.23776412, -0.275), (-0.20225424, -0.14694634, -0.275), (-0.25, -2.1855696e-8, -0.275), (-0.20225427, 0.1469463, -0.275), (-0.07725426, 0.23776412, -0.275), (0.07725424, 0.23776415, -0.275), (-0.07725427, -0.23776412, 0.275), (-0.07725427, -0.23776412, -0.275), (0.07725428, -0.23776412, -0.275), (0.07725428, -0.23776412, 0.275), (-0.25, -2.1855696e-8, 0.275), (-0.20225427, 0.1469463, 0.275), (-0.20225427, 0.1469463, -0.275), (-0.25, -2.1855696e-8, -0.275), (0.20225431, -0.14694624, 0.275), (0.07725428, -0.23776412, 0.275), (0.07725428, -0.23776412, -0.275), (0.20225431, -0.14694624, -0.275), (0.20225431, -0.14694624, 0.275), (0.20225431, -0.14694624, -0.275), (0.25, 0, -0.275), (0.25, 0, 0.275), (-0.20225424, -0.14694634, -0.275), (-0.07725427, -0.23776412, -0.275), (-0.07725427, -0.23776412, 0.275), (-0.20225424, -0.14694634, 0.275), (-0.20225424, -0.14694634, -0.275), (-0.20225424, -0.14694634, 0.275), (-0.25, -2.1855696e-8, 0.275), (-0.25, -2.1855696e-8, -0.275), (0.20225425, 0.14694631, -0.275), (0.07725424, 0.23776415, -0.275), (0.07725424, 0.23776415, 0.275), (0.20225425, 0.14694631, 0.275), (0.20225425, 0.14694631, -0.275), (0.20225425, 0.14694631, 0.275), (0.25, 0, 0.275), (0.25, 0, -0.275), (-0.07725426, 0.23776412, 0.275), (0.07725424, 0.23776415, 0.275), (0.07725424, 0.23776415, -0.275), (-0.07725426, 0.23776412, -0.275), (-0.07725426, 0.23776412, 0.275), (-0.07725426, 0.23776412, -0.275), (-0.20225427, 0.1469463, -0.275), (-0.20225427, 0.1469463, 0.275), (-0.07725426, 0.23776412, 0.275), (-0.20225427, 0.1469463, 0.275), (-0.25, -2.1855696e-8, 0.275), (-0.20225424, -0.14694634, 0.275), (-0.07725427, -0.23776412, 0.275), (0.07725428, -0.23776412, 0.275), (0.20225431, -0.14694624, 0.275), (0.25, 0, 0.275), (0.20225425, 0.14694631, 0.275), (0.07725424, 0.23776415, 0.275)]
        uniform token subdivisionScheme = "none"
    }
	
    quath[] orientations = [(-0.390625, 0.808594, -0.0162354, 0.439941), (0.769043, 0.274414, -0.00107861, 0.577148)]
    point3f[] positions = [(-1.7825685, 1.7859623, 5.1538763), (-1.5789804, 1.7658011, 5.060044)]
}
```

**Comment by asluk** [comment]
1. Open repo-prototypes.usda in usdView
2. See that the meshes are not drawn:
Warning: in _ComputeExtentAtTimePreamble at line 1024 of C:\Users\aluk\git\USD\pxr\usd\usdGeom\pointInstancer.cpp -- /pointinstancer -- no prototypes
Warning: in _Populate at line 134 of C:\Users\aluk\git\USD\pxr\usdImaging\usdImaging\pointInstancerAdapter.cpp -- Point instancer /pointinstancer does not have a valid 'prototypes' relationship. Not adding it to the render index.
3. In the interpreter, type:

stage = usdviewApi.stage
pi=stage.GetPrimAtPath('/pointinstancer')
proto = pi.GetProperty('prototypes')
protob = pi.GetProperty('prototypes_backup')
proto.SetTargets(protob.GetTargets())


4. Observe that nothing changes

**Comment by asluk** [comment]
# repro-positions.usda

```
#usda 1.0
(

)

def PointInstancer "pointinstancer"
{
    int[] protoIndices = [0, 1]
    
    rel prototypes = [
        </pointinstancer/mesh0>,
        </pointinstancer/mesh1>,
        ]
	
    def Mesh "mesh0"
    {
        uniform bool doubleSided = 1
        float3[] extent = [(-0.1, -0.09510565, -0.125), (0.1, 0.09510566, 0.125)]
        int[] faceVertexCounts = [3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3]
        int[] faceVertexIndices = [0, 2, 1, 0, 3, 2, 0, 4, 3, 0, 5, 4, 0, 6, 5, 0, 7, 6, 0, 8, 7, 0, 9, 8, 10, 12, 11, 10, 13, 12, 14, 16, 15, 14, 17, 16, 18, 20, 19, 18, 21, 20, 22, 24, 23, 22, 25, 24, 26, 28, 27, 26, 29, 28, 30, 32, 31, 30, 33, 32, 34, 36, 35, 34, 37, 36, 38, 40, 39, 38, 41, 40, 42, 44, 43, 42, 45, 44, 42, 46, 45, 42, 47, 46, 42, 48, 47, 42, 49, 48, 42, 50, 49, 42, 51, 50, 52, 54, 53, 52, 55, 54, 56, 58, 57, 56, 59, 58]
        point3f[] points = [(-0.0809017, -0.058778536, -0.125), (-0.1, -8.742278e-9, -0.125), (-0.080901705, 0.05877852, -0.125), (-0.030901704, 0.09510565, -0.125), (0.030901698, 0.09510566, -0.125), (0.080901705, 0.058778524, -0.125), (0.1, 0, -0.125), (0.080901735, -0.058778495, -0.125), (0.030901713, -0.09510565, -0.125), (-0.03090171, -0.09510565, -0.125), (-0.080901705, 0.05877852, -0.125), (-0.1, -8.742278e-9, -0.125), (-0.1, -8.742278e-9, 0.125), (-0.080901705, 0.05877852, 0.125), (-0.030901704, 0.09510565, 0.125), (0.030901698, 0.09510566, 0.125), (0.030901698, 0.09510566, -0.125), (-0.030901704, 0.09510565, -0.125), (-0.030901704, 0.09510565, 0.125), (-0.030901704, 0.09510565, -0.125), (-0.080901705, 0.05877852, -0.125), (-0.080901705, 0.05877852, 0.125), (0.080901705, 0.058778524, -0.125), (0.030901698, 0.09510566, -0.125), (0.030901698, 0.09510566, 0.125), (0.080901705, 0.058778524, 0.125), (0.080901705, 0.058778524, -0.125), (0.080901705, 0.058778524, 0.125), (0.1, 0, 0.125), (0.1, 0, -0.125), (0.080901735, -0.058778495, 0.125), (0.030901713, -0.09510565, 0.125), (0.030901713, -0.09510565, -0.125), (0.080901735, -0.058778495, -0.125), (0.080901735, -0.058778495, 0.125), (0.080901735, -0.058778495, -0.125), (0.1, 0, -0.125), (0.1, 0, 0.125), (-0.0809017, -0.058778536, -0.125), (-0.0809017, -0.058778536, 0.125), (-0.1, -8.742278e-9, 0.125), (-0.1, -8.742278e-9, -0.125), (-0.03090171, -0.09510565, 0.125), (0.030901713, -0.09510565, 0.125), (0.080901735, -0.058778495, 0.125), (0.1, 0, 0.125), (0.080901705, 0.058778524, 0.125), (0.030901698, 0.09510566, 0.125), (-0.030901704, 0.09510565, 0.125), (-0.080901705, 0.05877852, 0.125), (-0.1, -8.742278e-9, 0.125), (-0.0809017, -0.058778536, 0.125), (-0.03090171, -0.09510565, 0.125), (-0.0809017, -0.058778536, 0.125), (-0.0809017, -0.058778536, -0.125), (-0.03090171, -0.09510565, -0.125), (-0.03090171, -0.09510565, 0.125), (-0.03090171, -0.09510565, -0.125), (0.030901713, -0.09510565, -0.125), (0.030901713, -0.09510565, 0.125)]
        uniform token subdivisionScheme = "none"
    }

    def Mesh "mesh1"
    {
        uniform bool doubleSided = 1
        float3[] extent = [(-0.25, -0.23776412, -0.275), (0.25, 0.23776415, 0.275)]
        int[] faceVertexCounts = [3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3]
        int[] faceVertexIndices = [0, 2, 1, 0, 3, 2, 0, 4, 3, 0, 5, 4, 0, 6, 5, 0, 7, 6, 0, 8, 7, 0, 9, 8, 10, 12, 11, 10, 13, 12, 14, 16, 15, 14, 17, 16, 18, 20, 19, 18, 21, 20, 22, 24, 23, 22, 25, 24, 26, 28, 27, 26, 29, 28, 30, 32, 31, 30, 33, 32, 34, 36, 35, 34, 37, 36, 38, 40, 39, 38, 41, 40, 42, 44, 43, 42, 45, 44, 46, 48, 47, 46, 49, 48, 50, 52, 51, 50, 53, 52, 50, 54, 53, 50, 55, 54, 50, 56, 55, 50, 57, 56, 50, 58, 57, 50, 59, 58]
        point3f[] points = [(0.20225425, 0.14694631, -0.275), (0.25, 0, -0.275), (0.20225431, -0.14694624, -0.275), (0.07725428, -0.23776412, -0.275), (-0.07725427, -0.23776412, -0.275), (-0.20225424, -0.14694634, -0.275), (-0.25, -2.1855696e-8, -0.275), (-0.20225427, 0.1469463, -0.275), (-0.07725426, 0.23776412, -0.275), (0.07725424, 0.23776415, -0.275), (-0.07725427, -0.23776412, 0.275), (-0.07725427, -0.23776412, -0.275), (0.07725428, -0.23776412, -0.275), (0.07725428, -0.23776412, 0.275), (-0.25, -2.1855696e-8, 0.275), (-0.20225427, 0.1469463, 0.275), (-0.20225427, 0.1469463, -0.275), (-0.25, -2.1855696e-8, -0.275), (0.20225431, -0.14694624, 0.275), (0.07725428, -0.23776412, 0.275), (0.07725428, -0.23776412, -0.275), (0.20225431, -0.14694624, -0.275), (0.20225431, -0.14694624, 0.275), (0.20225431, -0.14694624, -0.275), (0.25, 0, -0.275), (0.25, 0, 0.275), (-0.20225424, -0.14694634, -0.275), (-0.07725427, -0.23776412, -0.275), (-0.07725427, -0.23776412, 0.275), (-0.20225424, -0.14694634, 0.275), (-0.20225424, -0.14694634, -0.275), (-0.20225424, -0.14694634, 0.275), (-0.25, -2.1855696e-8, 0.275), (-0.25, -2.1855696e-8, -0.275), (0.20225425, 0.14694631, -0.275), (0.07725424, 0.23776415, -0.275), (0.07725424, 0.23776415, 0.275), (0.20225425, 0.14694631, 0.275), (0.20225425, 0.14694631, -0.275), (0.20225425, 0.14694631, 0.275), (0.25, 0, 0.275), (0.25, 0, -0.275), (-0.07725426, 0.23776412, 0.275), (0.07725424, 0.23776415, 0.275), (0.07725424, 0.23776415, -0.275), (-0.07725426, 0.23776412, -0.275), (-0.07725426, 0.23776412, 0.275), (-0.07725426, 0.23776412, -0.275), (-0.20225427, 0.1469463, -0.275), (-0.20225427, 0.1469463, 0.275), (-0.07725426, 0.23776412, 0.275), (-0.20225427, 0.1469463, 0.275), (-0.25, -2.1855696e-8, 0.275), (-0.20225424, -0.14694634, 0.275), (-0.07725427, -0.23776412, 0.275), (0.07725428, -0.23776412, 0.275), (0.20225431, -0.14694624, 0.275), (0.25, 0, 0.275), (0.20225425, 0.14694631, 0.275), (0.07725424, 0.23776415, 0.275)]
        uniform token subdivisionScheme = "none"
    }
	
    quath[] orientations = [(-0.390625, 0.808594, -0.0162354, 0.439941), (0.769043, 0.274414, -0.00107861, 0.577148)]
    point3f[] positions = []
    point3f[] positions_backup = [(-1.7825685, 1.7859623, 5.1538763), (-1.5789804, 1.7658011, 5.060044)]
}
```

**Comment by asluk** [comment]
1. Open repo-positions.usda in usdView
2. See that the meshes remain at the origin
3. In the interpreter, type:

stage = usdviewApi.stage
pi=stage.GetPrimAtPath('/pointinstancer')
pos = pi.GetProperty('positions')
posb = pi.GetProperty('positions_backup')
pos.Set(posb.Get())

4. Observe that nothing changes, and in the console there's an error message:

Error in 'pxrInternal_v0_19__pxrReserved__::HdStVBOMemoryManager::_StripedBufferArrayRange::CopyData' at line 649 in file C:\b\w\da639afa0455b478\USD\pxr\imaging\lib\hdSt\vboMemoryManager.cpp : 'Failed verification: ' (VBO && VBO->GetId()) ' -- VBO doesn't exist for translate'

**Comment by jilliene** [comment]
Filed as internal issue #USD-5774

**Comment by asluk** [comment]
closed in v20.05 with https://github.com/PixarAnimationStudios/USD/commit/29d6803a625344fb9b7dae59c0278e52bf74271c
